invokeai/app/services/model_install/model_install_default.py

```diff
@@ -46,15 +63,30 @@ class ModelInstallService(ModelInstallServiceBase):
     _event_bus: Optional[EventServiceBase] = None
     _install_queue: Queue[ModelInstallJob]
     _install_jobs: List[ModelInstallJob]
+    _running: bool
     _logger: Logger
+    _lock: threading.Lock
+    _stop_event: threading.Event
     _cached_model_paths: Set[Path]
+    _session: Optional[
+        Session
+    ] = None  # This can be a customized requests.Session, for use during debugging or when internet not available.
+    _download_queue: DownloadQueueServiceBase
+    _downloads_changed_event: threading.Event
     _models_installed: Set[str]
+    _metadata_store: ModelMetadataStore
+    _metadata_cache: Dict[ModelSource, Optional[AnyModelRepoMetadata]]
+    _download_cache: Dict[AnyHttpUrl, ModelInstallJob]
+    _next_job_id: int = 0
```

Comment:
The class variables have been removed. I've also removed similar constructs from other files in this PR. Let me know if I missed any. It gets confusing because Pydantic classes do use these class variables. For what it's worth, you'll find similar static class declarations in multiple places in the InvokeAI code that were written by other members of the team. Here's one picked at random: `invokeai/app/services/board_images/board_image_records_sqlite.py`. 

Funny enough, last summer I asked some of the team members on Discord about this coding style and was assured that it was the pythonic way to do it. Maybe the style is coming from the way that class variables are used in Pydantic?

Reply:
Ya, I've noticed that we have a habit of including them.

I hadn't though about it much, but I think you could be right that Pydantic could be encouraging this style (plus it's tempting for anyone coming from a strictly typed language). Pydantic is of course using this mechanism for the meta-programming magic that they do - so very different than a vanilla Python class.